File: frontend/src/utils/notification.ts
import type { NotificationItem } from '../types'

// 通知只存类型与目标，动词文案在前端统一维护
export function describeNotification(item: NotificationItem): string {
  switch (item.type) {
    case 'LIKE':
      return item.targetType === 'PHOTO' ? '赞了你的照片' : '赞了你的相册'
    case 'COMMENT':
      return '评论了你的相册'
    case 'REPLY':
      return '回复了你的评论'
    case 'FOLLOW':
      return '关注了你'
    case 'MENTION':
      return '在评论里提到了你'
    default:
      return '有新动态'
  }
}

// 关注类通知跳到关注者主页，其余回到被操作的相册
export function notificationLink(item: NotificationItem): string | null {
  if (item.type === 'FOLLOW') {
    return item.actorId ? `/user/${item.actorId}` : null
  }
  const albumId = item.targetAlbumId ?? (item.targetType === 'ALBUM' ? item.targetId : null)
  return albumId ? `/album/${albumId}` : null
}

File: backend/src/main/java/com/photoshare/backend/entity/Notification.java
package com.photoshare.backend.entity;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class Notification {

    public static final String TYPE_LIKE = "LIKE";

    public static final String TYPE_COMMENT = "COMMENT";

    public static final String TYPE_REPLY = "REPLY";

    public static final String TYPE_FOLLOW = "FOLLOW";

    public static final String TYPE_MENTION = "MENTION";

    public static final String TARGET_ALBUM = "ALBUM";

    public static final String TARGET_USER = "USER";

    private String id;

    private String userId;

    private String actorId;

    private String type;

    private String targetType;

    private String targetId;

    private String snippet;

    private Boolean isRead;

    private LocalDateTime createdAt;

    // 以下字段由查询关联得到，不写入通知表
    private String actorName;

    private String actorAvatarUrl;

    private String targetTitle;

    private String targetAlbumId;
}
